### PR TITLE
User can add activities

### PR DESCRIPTION
**Pull request by sundqvistjohan**
[PT - User can add activities](https://www.pivotaltracker.com/story/show/171051386)

_As a Tripwiz user
In order to get a custom tailored trip
I need to be able to choose the activities I'm interested in_

Adds webmock and stubs out old and new external API requests
Adds ActivityType and Activity models, ActivityType has many Activities
Adds ActivityTypeController with create action and route, the create action invokes creation of Activities through class methods in the Activity model. It responds, if successfull, with the created activities.

### CODE REVIEW

File: app/controllers/api/v1/activity_types_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class Api::V1::ActivityTypesController < ApplicationController
+  def create
+    number_of_activities = permitted_params.require(:actTimes)
+    activity_type = ActivityType.create(activity_type: permitted_params.require(:activity_type),
+                                        trip_id: permitted_params.require(:trip))
```
Comment (tochman) [inline app/controllers/api/v1/activity_types_controller.rb]: Why are you doing this this way? I don't understand the upside of permitting it here...

**Review by tochman — COMMENTED** [review]
TBH, I don't really understand what is going on here. The naming puts me off a bit....

File: app/controllers/api/v1/activity_types_controller.rb (hunk 1 above)
Comment (sundqvistjohan) [reply app/controllers/api/v1/activity_types_controller.rb]: yeah it really makes no sense as I'm not mass-assigning them...

Comment (sundqvistjohan) [reply app/controllers/api/v1/activity_types_controller.rb]: The naming puts me off as well but I've spent way to much time trying to come up with less shitty names

Comment (sundqvistjohan) [reply app/controllers/api/v1/activity_types_controller.rb]: Screenshot 2020-02-07 at 14 20 57

Dunno if this makes it clearer

**Comment by mantanquzzmo** [comment]
lgtm

**Pull request merged**